src/wallet/signer.ts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable import/max-dependencies -- These imports are required for Signer */
+import { BigNumber } from "bignumber.js";
+import { flatMap } from "lodash";
+import { decodeAccountID } from "ripple-address-codec";
+import {
+  encode,
+  decode,
+  encodeForSigning,
+  encodeForSigningClaim,
+} from "ripple-binary-codec";
+import { JsonObject } from "ripple-binary-codec/dist/types/serialized-type";
+import {
+  sign as signWithKeypair,
+  verify as verifySignature,
+} from "ripple-keypairs";
+
+import { ValidationError } from "../common/errors";
+import { SignedTransaction } from "../common/types/objects";
+import { Signer } from "../models/common";
+import { Transaction } from "../models/transactions";
+import { verifyBaseTransaction } from "../models/transactions/common";
+import { computeBinaryTransactionHash } from "../utils";
+import Wallet from "../Wallet";
+
+/**
+ * Uses a wallet to cryptographically sign a transaction which proves the owner of the wallet
+ * is issuing this transaction.
+ *
+ * @param wallet - A Wallet that holds your cryptographic keys.
+ * @param tx - The Transaction that is being signed.
+ * @returns A signed Transaction and a Transaction id that corresponds to the pre-signed Transaction.
+ */
+function sign(wallet: Wallet, tx: Transaction): SignedTransaction {
+  return wallet.signTransaction(tx, { signAs: "" });
+}
+
+/**
+ * Takes several transactions (in object or blob form) and creates a single transaction with all Signers
+ * that then gets signed and returned.
+ *
+ * @param transactions - An array of Transactions (in object or blob form) to combine and sign.
+ * @returns A single signed Transaction which has all Signers from transactions within it.
+ * @throws ValidationError if:
+ * - There were no transactions given to sign
+ * - The SigningPubKey field is not the empty string in any given transaction
+ * - Any transaction is missing a Signers field.
+ */
+function multisign(
+  transactions: Array<Transaction | string>
+): SignedTransaction {
+  if (transactions.length === 0) {
+    throw new ValidationError("There were 0 transactions given to multisign");
+  }
+
+  transactions.forEach((txOrBlob) => {
+    const tx: Transaction = getDecodedTransaction(txOrBlob);
+
+    // This will throw a more clear error for JS users if any of the supplied transactions has incorrect formatting
+    // TODO: Replace this with verify() (The general validation function for all Transactions)
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- verify does not accept Transaction type
+    verifyBaseTransaction(tx as unknown as Record<string, unknown>);
```

Comment:
This should directly cast to `Record<string, unknown>` (and that linting error can be disabled)

Reply:
Converting to unknown is how to tell Typescript that you meant to do a conversion which it doesn't see any overlap with. (It unfortunately won't compile if I directly convert, or if I try to implicitly let it convert by not casting at all)